src/components/fields/ObjectField.js

```diff
@@ -73,6 +74,13 @@ class ObjectField extends Component {
     const { definitions, fields, formContext } = registry;
     const { SchemaField, TitleField, DescriptionField } = fields;
     const schema = retrieveSchema(this.props.schema, definitions, formData);
+    const objectIdSchema = toIdSchema(
+      schema,
+      null,
+      definitions,
+      formData,
+      idSchema["$id"]
+    );
```

Comment:
@redixhumayun you can use or adapt the test of `ObjetField` from my branch: https://github.com/mozilla-services/react-jsonschema-form/compare/master...jonathonlui:fix-nested-object-dynamic-conditionals

When `ObjectField.js` does not have these changes applied then using `npm test` the test correctly fails . However, there is an issue with the test when running `npm run tdd`: the first test run will correctly fail but when the test is rerun because of a changed file, the test will incorrectly pass. I'm guessing it's related to resetting the `console.error` stub but I was not able to resolve the issue.

Reply:
@edi9999 - Hi added a failing and passing test to ObjectField_test.js. Take a look when you get a chance.